File: oa-common/src/main/java/com/nytdacm/oa/entity/Article.java
package com.nytdacm.oa.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "t_articles")
@Getter
@Setter
@AllArgsConstructor
public class Article extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "article_id", nullable = false, updatable = false)
    private Long articleId;

    private String title;

    @Column(columnDefinition = "text")
    private String content;

    @ManyToOne
    private User author;

    private Boolean published = false;
}

File: oa-common/src/main/java/com/nytdacm/oa/entity/Article.kt
package com.nytdacm.oa.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "t_articles")
class Article(
    var title: String,
    @Column(columnDefinition = "text") var content: String,
    @ManyToOne var author: User,
    var published: Boolean = false,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "article_id")
    var articleId: Long? = null,
) : BaseEntity()
